src/main/java/com/flightstats/hub/dao/aws/S3Verifier.java

```diff
@@ -152,18 +152,25 @@ protected void verifyChannel(VerifierRange range) {
         String channelName = range.getChannelConfig().getDisplayName();
         SortedSet<ContentKey> keysToAdd = getMissing(range.getStartPath(), range.getEndPath(), channelName, s3SingleContentDao, new TreeSet<>());
         logger.debug("verifyChannel.starting {}", range);
+        MinutePath lastCompleted = range.getEndPath();
         for (ContentKey key : keysToAdd) {
             logger.trace("found missing {} {}", channelName, key);
             metricsService.increment(MISSING_ITEM_METRIC_NAME);
             boolean success = s3WriteQueue.add(new ChannelContentKey(channelName, key));
             if (!success) {
                 logger.error("unable to queue missing item {} {}", channelName, key);
                 metricsService.increment(VERIFIER_FAILED_METRIC_NAME);
-                return;
+                lastCompleted = new MinutePath(key.getTime().minusMinutes(1));
+                break;
```

Comment:
I don't recall if lastCompleted as startPath becomes inclusive or exclusive in the range? If it is inclusive couldn't we set the last completed to key.getTime without the minus 1? 
Either way I think this will be helpful

Reply:
I don't recall either, but assumed it would return stuff starting with the beginning of the startPath minute. I'll check it out.

Reply:
It is probably safer the way you did it anyway to avoid minute boundary issues now that I think about it

Reply:
Follow up after Paul and I looked at the code:
- Spoke write DAO startPath is exclusive (nothing from start minute will be returned, so it assumes LAST_COMPLETED was successful)
- S3SingleContentDao startPath is inclusive (will return stuff from start minute.

So they aren't consistent. This is fine for now, I guess, since we're doing SPOKE KEYS - S3 KEYS, so if we get extra stuff from s3, we don't care. Means we're having to do more s3 queries, which is not great, but...whatever.

Reply:
So with that said, subtracting the minute from the failed item seems correct-ish.